src/control/security/pem.go

```diff
@@ -38,33 +38,43 @@ import (
 )
 
 const (
-	SafeKeyPerm  os.FileMode = 0400
-	SafeCertPerm os.FileMode = 0664
-	SafeDirPerm  os.FileMode = 0700
+	MaxKeyPerm  os.FileMode = 0700
+	MaxCertPerm os.FileMode = 0664
+	MaxDirPerm  os.FileMode = 0700
 )
 
-type insecureError struct {
+type improperError struct {
 	filename string
 	perms    os.FileMode
 }
 
-func (e *insecureError) Error() string {
-	return fmt.Sprintf("%s has insecure permissions set to %#o",
+func (e *improperError) Error() string {
+	return fmt.Sprintf("%s has incorrect permissions please set to %#o",
 		e.filename, e.perms)
 }
 
+func checkMaxPermissions(filePath string, mode os.FileMode, modeMax os.FileMode) error {
+	maxPermMask := (^modeMax.Perm()) & 0777
+	maskedPermissions := maxPermMask & mode.Perm()
+
+	if maskedPermissions != 0 {
+		return &improperError{filePath, modeMax}
```

Comment:
Would it make sense to include the actual mode perms in the error struct as well? That way both the actual and max perms could be displayed

Reply:
Indeed. Added it and modified the message to be what mike suggested.